common/blockchain_util.py

```diff
@@ -41,7 +41,13 @@ def read_contract_address(self, net_id, path, key):
         return Web3.toChecksumAddress(contract[str(net_id)][key])
 
     def contract_instance(self, contract_abi, address):
-        web3_object = Web3(self.provider)
+        if self._provider_type == "HTTP_PROVIDER":
+            provider = Web3.HTTPProvider(self._provider_url)
+        elif self._provider_type == "WS_PROVIDER":
+            provider = web3.providers.WebsocketProvider(self._provider_url)
+        web3_object = Web3(provider)
+        self.provider = provider
+        self.web3_object = web3_object
```

Comment:
@prashantramangupta why this change if this is happening in constructor.

Reply:
discussed with @keshrisohit. web3 connection was getting stale so to avoid as of now gone with the above approach.